### PR TITLE
Add discrete categories

### PR DESCRIPTION
**Pull request by barrettj12**
I've defined discrete categories using `Hom[ x , y ] = x ≡ y`.

I'm now trying to show that a function `f : A → ob C` determines a `Functor (Discrete A) C`, but I'm having trouble showing that this preserves composition. Maybe someone else can help.

### CODE REVIEW

**Comment by kangrongji** [comment]
I think you can use J-rule twice, so you only need to treat composition of identity morphisms.

**Comment by barrettj12** [comment]
Makes sense, but I'm not quite sure how to apply `J` here.

I think the problem might be that given `p : x ≡ y` and `q : y ≡ z`
- `F p` and `F (p ∙ q)` are obtained by transport from `id {F x}`
- `F q` is obtained from `id {F y}`

so it's not clear how to compose them.

File: Cubical/Categories/Instances/Discrete.agda [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+open import Cubical.Foundations.Transport
+
+private
+  variable
+    ℓ ℓC ℓC' : Level
+
+open Category
+
+-- Discrete category
+Discrete : hGroupoid ℓ → Category ℓ ℓ
```
Comment (mortberg) [inline Cubical/Categories/Instances/Discrete.agda:21]: Rename to `DiscreteCategory` ?

**Review by mortberg — CHANGES_REQUESTED** [review]
Just a name change suggestion

File: Cubical/Categories/Instances/Discrete.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/Categories/Instances/Discrete.agda:21]: My suggestion would be more consistent with https://github.com/agda/cubical/blob/master/Cubical/Categories/Instances/Rings.agda. If someone wants a shorter name they can rename when importing.

Comment (barrettj12) [reply Cubical/Categories/Instances/Discrete.agda:21]: Okay, will do.

**Review by mortberg — APPROVED** [review]

**Pull request merged**